packages/fast-tooling/src/web-components/css-box-model/css-box-model.ts

```diff
@@ -0,0 +1,459 @@
+import { provideFASTDesignSystem } from "@microsoft/fast-components";
+import { observable } from "@microsoft/fast-element";
+import { fastToolingUnitsTextField } from "../units-text-field";
+import {
+    CSSDeclarationDictionary,
+    mapCSSInlineStyleToCSSPropertyDictionary,
+} from "../../data-utilities/mapping.mdn-data";
+import { FormAssociatedCSSBoxModel } from "./css-box-model.form-associated";
+
+provideFASTDesignSystem()
+    .withPrefix("fast-tooling")
+    .register(fastToolingUnitsTextField());
+
+/**
+ * Regular expression for identifying the length part of a shorthand CSS statement.
+ * i.e. It will match the "1px" part of "1px solid red".
+ */
+const cssLengthRegExp: RegExp = new RegExp(
+    /^auto$|^thin$|^medium$|^thick$|^[+-]?[0-9]+\.?([0-9]+)?(ch|rem|vh|vw|vmin|vmax|px|em|ex|%|in|cm|mm|pt|pc)?$/
+);
+
+enum valueType {
+    shortQuad = "shortQuad",
+    singleVal = "singleVal",
+    shortMixed = "shortMixed",
+    singleOverride = "singleOverride",
```

Comment:
I'm going to take a stab at this, let me know if I interpreted it incorrectly:

```suggestion
    shorthandWithFourValues = "shorthand-with-four-values",
    oneValue = "one-value",
    shorthandWithMixedTypes = "shorthand-with-mixed-types",
    override = "override",
```

Reply:
Changed